Source/ASCollectionView.mm

```diff
@@ -761,6 +765,28 @@ - (void)setUsesSynchronousDataLoading:(BOOL)usesSynchronousDataLoading
   self.dataController.usesSynchronousDataLoading = usesSynchronousDataLoading;
 }
 
+- (void)invalidateFlowLayoutDelegateMetrics {
+  UICollectionViewLayout *layout = self.collectionViewLayout;
+  CGSize defaultSize = ({
+    UICollectionViewFlowLayout *flowLayout = ASDynamicCast(layout, UICollectionViewFlowLayout);
+    flowLayout ? flowLayout.itemSize : CGSizeZero;
+  });
+
+  for (ASCollectionElement *element in self.dataController.pendingMap.itemElements) {
+    ASCellNode *node = element.node;
+    if (node.shouldUseUIKitCell) {
```

Comment:
1. It's probably inconsequential, but maybe we should just ignore calls where the layout isn't a flow layout, rather than using size 0?

1. Instead of building a giant array – I don't really like the `itemElements` property and may propose removing it – it's better to enumerate the map directly and `if (element.supplementaryElementKind) { continue; }` Note to self: add a `UICollectionElementCategory elementCategory` property to ASCollectionElement to simplify this distinction.

1. Shouldn't we also refetch header/footer reference sizes here? Meshes well with 2

Reply:
@Adlai-Holler some merge conflicts have arisen, and I'm afraid my time availability is nearing a local minimum. :( Do you think this PR is acceptable to land as-is if the conflicts are fixed?

1. I think it's better to have the 0 default, but indeed maybe no one will ever know :)
2. The iteration suggestion sounds like an improvement for sure. I'm not sure I can commit to doing it only because I would have to get some test cases running to exercise the change.
3. This seems potentially important, and indeed might motivate me to do #2.

Could you give me an Accept if you are comfortable with this, and then if I do 2 or 3, I can land without waiting?

Reply:
@Adlai-Holler It took me a while, but I finally got around to implementing your advice — and went quite a bit further. Thanks for suggesting these things, as they are improvements.

In particular, I was intentionally skipping over handling the update of supplementary element sizes, which is not great. That's now fixed and the code is better factored.

If you or @nguyenhuy can give me a sanity check review soon, I'd appreciate that - would like to merge this in pretty soon.